src/town_cmd.cpp

```diff
@@ -3162,44 +3244,7 @@ static void UpdateTownGrowRate(Town *t)
 		return;
 	}
 
-	/**
-	 * Towns are processed every TOWN_GROWTH_TICKS ticks, and this is the
-	 * number of times towns are processed before a new building is built.
-	 */
-	static const uint16 _grow_count_values[2][6] = {
-		{ 120, 120, 120, 100,  80,  60 }, // Fund new buildings has been activated
-		{ 320, 420, 300, 220, 160, 100 }  // Normal values
-	};
-
-	int n = 0;
-
-	const Station *st;
-	FOR_ALL_STATIONS(st) {
-		if (DistanceSquare(st->xy, t->xy) <= t->cache.squared_town_zone_radius[0]) {
-			if (st->time_since_load <= 20 || st->time_since_unload <= 20) {
-				n++;
-			}
-		}
-	}
-
-	uint16 m;
-
-	if (t->fund_buildings_months != 0) {
-		m = _grow_count_values[0][min(n, 5)];
-	} else {
-		m = _grow_count_values[1][min(n, 5)];
-		if (n == 0 && !Chance16(1, 12)) return;
-	}
-
-	/* Use the normal growth rate values if new buildings have been funded in
-	 * this town and the growth rate is set to none. */
-	uint growth_multiplier = _settings_game.economy.town_growth_rate != 0 ? _settings_game.economy.town_growth_rate - 1 : 1;
-
-	m >>= growth_multiplier;
-	if (t->larger_town) m /= 2;
-
-	t->growth_rate = TownTicksToGameTicks(m / (t->cache.num_houses / 50 + 1));
-	t->grow_counter = min(t->growth_rate, t->grow_counter);
+	if (t->fund_buildings_months == 0 && CountActiveStations(t) == 0 && !Chance16(1, 12)) return;
```

Comment:
You've changed the logic for that `Chance16`. Previously, it would cause the update of `growth_rate` to be skipped, with your patch it is still recalculated. Is this intentional, and if yes, why?

Reply:
@michicc Sort of. This patch always keeps growth_rate up-to-date just to keep logic more straightforward. It's also a bit of preparation for #6378. But it shouldn't change anything in case of that chance thing, only side-effect I could think of is a bit different growth_counter handling when using gs to switch normal/custom growth rates but it's already different due to scaling and doesn't seem to be a bad change. Or did I miss something?